src/backing/client/grpc_client/proto_util_test.cc

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <tuple>
+
+#include <gmock/gmock.h>
+
+#include "absl/strings/str_cat.h"
+#include "google/cloud/kms/v1/resources.pb.h"
+#include "google/cloud/kms/v1/service.pb.h"
+#include "src/backing/client/digest_case.h"
+#include "src/backing/client/grpc_client/proto_util.h"
+#include "src/backing/status/status.h"
+
+namespace kmsengine {
+namespace backing {
+namespace {
+
+using ::testing::Combine;
+using ::testing::StrEq;
+using ::testing::Values;
+using ::testing::ValuesIn;
+
+// Gets the underlying `bytes` attached to a `Digest` protobuf.
+std::string GetDigestBytes(google::cloud::kms::v1::Digest digest) {
+  switch (digest.digest_case()) {
+    case google::cloud::kms::v1::Digest::DigestCase::kSha256:
+      return digest.sha256();
+    case google::cloud::kms::v1::Digest::DigestCase::kSha384:
+      return digest.sha384();
+    case google::cloud::kms::v1::Digest::DigestCase::kSha512:
+      return digest.sha512();
+    default:
+      return "";
+  }
+}
+
+// Sample digests for testing purposes.
+const std::string kSampleDigests[] = {
+  // Example SHA-256 digest of "hello world" for testing.
+  "b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9",
+  "an arbitrary digest",
+  "",
+};
+
+class MakeDigestTest : public
+    testing::TestWithParam<std::tuple<DigestCase, std::string>> {
+  // Purposely empty; no fixtures to instantiate.
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    DigestParameters, MakeDigestTest,
+    Combine(Values(DigestCase::kSha256,
+                   DigestCase::kSha384,
+                   DigestCase::kSha512),
+            ValuesIn(kSampleDigests)));
+
+TEST_P(MakeDigestTest, MakeDigest) {
+  auto expected_digest_type = std::get<0>(GetParam());
+  auto expected_digest_bytes = std::get<1>(GetParam());
+
+  auto actual = MakeDigest(expected_digest_type, expected_digest_bytes);
+  EXPECT_EQ(FromProtoToDigestCase(actual.digest_case()), expected_digest_type);
+  EXPECT_THAT(GetDigestBytes(actual), StrEq(expected_digest_bytes));
+}
+
+// Mapping between `StatusCode` cases and their protobuf equivalents.
+struct CorrespondingStatusCode {
+  grpc::StatusCode proto;
+  StatusCode code;
+} kStatusCodeMapping[]{
```

Comment:
Nit: separate type definition from variable declaration as discussed previously

Reply:
Handling in #77.